addons/binding/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/handler/TrackerHandler.java

```diff
@@ -0,0 +1,356 @@
+        super(thing);
+
+        this.notificationBroker = notificationBroker;
+        this.notificationHandler = new NotificationHandler();
+        this.regions = regions;
+        this.sysLocation = sysLocation;
+        this.unitProvider = unitProvider;
+
+        trackerId = ConfigHelper.getTrackerId(thing.getConfiguration());
+        notificationBroker.registerHandler(trackerId, notificationHandler);
+
+        logger.debug("Tracker handler created: {}", trackerId);
+    }
+
+    /**
+     * Returns tracker id configuration of the thing.
+     *
+     * @return Tracker id
+     */
+    public String getTrackerId() {
+        return trackerId;
+    }
+
+    @Override
+    public void initialize() {
+        if (sysLocation != null) {
+            createBasicDistanceChannel();
+        } else {
+            logger.debug("System location is not set. Skipping system distance channel setup.");
+        }
+
+        mapDistanceChannels();
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * Create distance channel for measuring the distance between the tracker and the szstem.
+     */
+    private void createBasicDistanceChannel() {
+        @Nullable ThingHandlerCallback callback = getCallback();
+        if (callback != null) {
+            //find the system distance channel
+            ChannelUID systemDistanceChannelUID = new ChannelUID(thing.getUID(), CHANNEL_DISTANCE_SYSTEM_ID);
+            Channel systemDistance = thing.getChannel(CHANNEL_DISTANCE_SYSTEM_ID);
+            ChannelBuilder channelBuilder = null;
+            if (systemDistance != null) {
+                if (!systemDistance.getConfiguration().get(CONFIG_REGION_CENTER_LOCATION).equals(sysLocation.toFullString())) {
+                    logger.trace("Existing distance channel for system. Changing system location config parameter: {}", sysLocation.toFullString());
+
+                    channelBuilder = callback.editChannel(thing, systemDistanceChannelUID);
+                    Configuration configToUpdate = systemDistance.getConfiguration();
+                    configToUpdate.put(CONFIG_REGION_CENTER_LOCATION, sysLocation.toFullString());
+                    channelBuilder.withConfiguration(configToUpdate);
+                } else {
+                    logger.trace("Existing distance channel for system. No change.");
+                }
+            } else {
+                logger.trace("Creating missing distance channel for system.");
+
+                Configuration config = new Configuration();
+                config.put(ConfigHelper.CONFIG_REGION_NAME, CHANNEL_DISTANCE_SYSTEM_NAME);
+                config.put(CONFIG_REGION_CENTER_LOCATION, sysLocation.toFullString());
+                config.put(ConfigHelper.CONFIG_REGION_RADIUS, CHANNEL_DISTANCE_SYSTEM_RADIUS);
+
+                channelBuilder = callback.createChannelBuilder(systemDistanceChannelUID, CHANNEL_TYPE_DISTANCE)
+                        .withLabel("System Distance")
```

Comment:
I am wondering if it might be more user friendly to call it "Home Distance". We can keep it being called "system" technically internally. Wdyt?

Reply:
I wouldn't change it. This "Home" is not always home but it's always the openHAB system. BTW on Item level users can rename it to reflect the reality.

Reply:
ok